consensus/bor/span_store.go

```diff
@@ -0,0 +1,63 @@
+package bor
+
+import (
+	"context"
+
+	"github.com/ethereum/go-ethereum/consensus/bor/heimdall/span"
+	"github.com/ethereum/go-ethereum/log"
+	lru "github.com/hashicorp/golang-lru"
+)
+
+// SpanStore acts as a simple middleware to cache span data populated from heimdall. It is used
+// in multiple places of bor consensus for verification.
+type SpanStore struct {
+	store          *lru.ARCCache
+	heimdallClient IHeimdallClient
+}
+
+func NewSpanStore(heimdallClient IHeimdallClient) SpanStore {
+	cache, _ := lru.NewARC(10)
+	return SpanStore{
+		store:          cache,
+		heimdallClient: heimdallClient,
+	}
+}
+
+const (
+	defaultSpanLength = 6400 // Default span length i.e. number of bor blocks in a span
+	zerothSpanEnd     = 255  // End block of 0th span
```

Comment:
Although the default span size of the first span is 256, it is possible to configure it through [contract genesis script](https://github.com/maticnetwork/genesis-contracts/blob/master/generate-borvalidatorset.js#L15-L19). Can this value be read from genesis contract itself?

Reply:
Good point, will fetch it from contract on the first run. Thanks!

Reply:
Updated the logic in https://github.com/maticnetwork/bor/pull/1472/commits/fc2462d0ee52c0bbd1bd5a686c5c3873f24584c9 as discussed today to use existing span's start and end block (as done in contracts) instead of relying on this formula with hardcoded values.